plugins/metalsmith-hierarchy.js

```diff
@@ -85,22 +77,23 @@ const findMeta = ({ children }, key, value) =>
     )
     .concat(...children.map((c) => findMeta(c, key, value)));
 
+const paths = {};
```

Comment:
pulling out paths to be in "module-scope" is the actual trick of having things cached here.